frontend/pages/DetailedViewPage.js

```diff
@@ -1,25 +1,35 @@
 import React, {Component} from 'react';
 import {searchByUPC} from "../utils/api";
-import { View, ActivityIndicator, StyleSheet} from 'react-native';
-
+import {View, ActivityIndicator, StyleSheet, FlatList} from 'react-native';
+import _ from 'lodash'
 import { Image, PricingCard, Overlay, Text, Input, Button } from 'react-native-elements';
 
-
 export default class DetailedViewPage extends Component {
-
     constructor(props) {
         super(props);
+        store = ''
+        //Had to wrap this in a try catch for when the component is loaded with no navigation (testing)
+        try {
+            store = this.props.navigation.state.params.store
+        } catch(err) {
+            console.log(err)
+        }
         this.state = {
             isLoading:true,
             priceOverlayVisible: false,
-            newPrice: ''
+            newPrice: '',
+            store: store
         }
     }
 
-
     async componentDidMount() {
         let data = await searchByUPC(this.props.navigation.state.params.upc);
-        this.setState({isLoading: false, ...data[0]})
+
+        //we only want the prices for the store we selected
+        oldPrices = _.find(data[0].stores, ['name', this.state.store]).prices
+
+        //add oldprices, as well as other item data, and let the app load
+        this.setState({isLoading: false, oldPrices: oldPrices, ...data[0]})
```

Comment:
I'm not sure how I feel about unrolling this data list. Perhaps at some point the api call you're making may return differently structured data. I'd rather you simply do something like:
```data: data[0]```
for example, since it is simpler to catch errors that may arise if the backend changes the format the api returns data. Also, **if the server returns an error** (Which may very well happen), you're code currently does not handle that edge case. You need someway of checking the status code of the response you get in searchByUPC

Reply:
cleaned up